### PR TITLE
Hand tracking

### PR DESCRIPTION
**Pull request by NathanGillispie**
Code from today's meeting Jan 28.

Added a "Next Knot" button which increments the knot in VR.
Additionally, there are a few cubes in the back which can be deleted by pointing at them and pinching.

### CODE REVIEW

File: Assets/Scripts/ButtonKnotIncrementor.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        if (crossingNumber >= numberOfKnots.Length)  crossingNumber = 3;
+
+        knotGameObject.crossingNumber = crossingNumber;
+        knotGameObject.ordering = ordering;
+    }
+
+    
+    // void Update()
```
Comment (16dprice) [inline Assets/Scripts/ButtonKnotIncrementor.cs]: We don't want commented code in our codebase, so I recommend getting rid of this function if we don't need it for the game to work properly

File: Assets/Scripts/ButtonListener.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public UnityEvent defaultEvent;
+
+    void Start()
+    {
+        GetComponent<ButtonController>().InteractableStateChanged.AddListener(InitiateEvent);
+    }
+
+    void InitiateEvent(InteractableStateArgs state)
+    {
+        if (state.NewInteractableState == InteractableState.ProximityState)
```
Comment (16dprice) [inline Assets/Scripts/ButtonListener.cs]: this would probably be better as a switch statement

**Review by 16dprice — CHANGES_REQUESTED** [review]

File: Assets/Scripts/ButtonKnotIncrementor.cs (hunk 1 above)
Comment (NathanGillispie) [reply Assets/Scripts/ButtonKnotIncrementor.cs]: I removed this

File: Assets/Scripts/ButtonListener.cs (hunk 2 above)
Comment (NathanGillispie) [reply Assets/Scripts/ButtonListener.cs]: I removed this file

**Review by 16dprice — APPROVED** [review]

**Pull request merged**